observability/metrics/src/main/scala/com/daml/metrics/api/MetricHandle.scala

```diff
@@ -17,7 +17,11 @@ trait MetricHandle {
 
 object MetricHandle {
 
-  trait Factory {
+  trait LabeledMetricsFactory {
+    this: MetricsFactory =>
```

Comment:
Any concrete reasons to not directly extending `MetricFactory` ?
```scala
trait LabeledMetricFactory extends MetricFactory
```

I think the construction you used allow more flexibility if we were to add more traits like it, but it is unlikely we need it. And it makes referencing the type more complex. Just `LabeledMetricsFactory` vs `MetricsFactory with LabeledMetricsFactory`.

Reply:
Good point, I wanted to differentiate between them with the tagging but your idea makes more sense. Will simplify it